### PR TITLE
Add delete board functionality

### LINKED ISSUE #25: `delete_board()`

### PR DESCRIPTION
**Pull request by jessat**
Closes #25 

Related to #3 

Added a delete_board() method and its corresponding test function

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <img src="./assets/pytrello2.png" width="160" height="160" align="right">
 
 [![CI
-Python](https://github.com/5-jigglypuff/pytrello2/actions/workflows/ci-tests.yml/badge.svg?branch=master)](https://github.com/5-jigglypuff/pytrello2/actions/workflows/ci-tests.yml)
+Python](https://github.com/5-jigglypuff/pytrello2/actions/workflows/tests.yml/badge.svg?branch=master)](https://github.com/5-jigglypuff/pytrello2/actions/workflows/tests.yml)
```
Comment (vchrombie) [inline README.md:6]: Do you think we can open a different PR for this change?

Let's maintain the scope of this PR to add the two functions for board.

WDYT?

File: tests/test_board.py [hunk 2]
```diff
@@ -62,3 +62,13 @@ def test_create_board(mock_http_client):
...
+
+# Test for delete_board method
+def test_delete_board(mock_http_client):
+    board_data = load_mock_data(BOARD_MOCK_DATA)
+    mock_http_client.get.return_value = board_data
+                              
+    board_manager = BoardManager(mock_http_client)
+    board = board_manager.delete_board("test_board_id")
+    
+    assert isinstance(board, Board)
```
Comment (vchrombie) [inline tests/test_board.py:74]: You can add a new line at the end of this file.

File: pytrello2/board.py [hunk 3]
```diff
@@ -35,10 +35,17 @@ def get_all_boards(self):
 
     def create_board(self, name, desc):
         """
-        Returns a Card object with the given ID.
+        Returns a Board object with the given ID.
```
Comment (vchrombie) [inline pytrello2/board.py:38]: Nice catch, but the doc string should explain what the function is about. Maybe this is not the correct explanation.

WDYT?

File: pytrello2/board.py [hunk 4]
```diff
@@ -35,10 +35,17 @@ def get_all_boards(self):
...
         """
         data = {}
         data["name"] = name
         data["desc"] = desc
         board_data = self.client.post("boards/", data)
         return Board(board_data)
+
+    def delete_board(self, board_id):
+        """
+        Returns a Board object with the given ID.
```
Comment (vchrombie) [inline pytrello2/board.py:48]: The doc string should explain what the function is about. Could we say something like deletes a board?

WDYT?

**Review by vchrombie — CHANGES_REQUESTED** [review]
I left a few comments.

Also, you can add the DCO sign off in the commit description not the commit message.

You can make the changes and amend it to the same commit.
```
git add .
git commit --amend -s
```

The first line of a commit is the commit message and description starts from the second line.
```
Add delete board functionality

This commit adds the delete board functionality.
The tests are also updated accordingly.

Signed-off-by: Jessie Theisen <jt1321@nyu.edu>
```

Please let me know if you need any help.

File: README.md (hunk 1 above)
Comment (jessat) [reply README.md:6]: Sounds good, I'll add a new PR for the readme.

File: pytrello2/board.py (hunk 3 above)
Comment (jessat) [reply pytrello2/board.py:38]: Agreed, will change it.

File: pytrello2/board.py (hunk 4 above)
Comment (jessat) [reply pytrello2/board.py:48]: Agreed, will change it.

**Comment by jessat** [comment]
@vchrombie Can you take a look?  I think I fixed the spacing issues, but I must have signed off incorrectly somewhere, sorry about that.

**Review by vchrombie — COMMENTED** [review]
The changes look good to me.

Can you fix the sign off so that we can merge it?
```
git checkout 3-implement-board-resource-methods
git reset --soft HEAD~4
git commit -s
git push origin 3-implement-board-resource-methods -f
```

@jessat, let me know if you need any help

**Comment by jessat** [comment]
Ok, done.

**Comment by vchrombie** [comment]
Good work on the sign off, a last change could be fixing the commit message itself
```bash
git checkout 3-implement-board-resource-methods
git commit --amend
```

and have your commit message something like below (you can let the sign off stay like that)
```txt
Add delete board functionality

This commit adds the delete board functionality.
The tests are also updated accordingly.
```

and then force push
```bash
git push origin 
git push origin 3-implement-board-resource-methods -f
```

@jessat, let me know if you need any help.

**Review by vchrombie — APPROVED** [review]
LGTM
